docs(book): initial version (#157)
ntls-io/nautilus-wallet: docs/book/src/design/wallet.md

# Wallet

> This provides the functionality of a non-custodial wallet to end-users without smartphones, accessible through a smartphone-based connector app with limited trust.

Reviewer: This requires some elaboration.  It is not immediately clear how an application can be both "smartphone-based" while also targeting users *without* smartphones.  I assume this is where the "limited trust" comes in but there is value in being explicit here.

Author: How about something like this?

```suggestion
This provides the functionality of a non-custodial wallet to end-users who don't own a smartphone themselves, accessible through a smartphone-based connector app with limited trust.
```

Reviewer: I think that is much clearer yes!
